### PR TITLE
Create testing scripts via tmux

### PR DESCRIPTION
**Pull request by corylanou**
All wip.  nothing to see yet.

### CODE REVIEW

**Comment by corylanou** [comment]
@otoolep This currently works, but I'm not going to take this any further.  I am going to open another PR and start an integration test suite that runs in Go.  Once that is in shape, it will make sense to bring this back to life.  I would like to collapse it as it has some fixes to the code that was discovered during this testing process.  Thoughts?

**Comment by otoolep** [comment]
Makes sense.  Are you proposing to close this PR? There are changes in there, that look like a short-term fix for the node-restart issue.

**Comment by corylanou** [comment]
Yes, I am proposing merging this into the master branch.

File: cmd/influxd/config.go [hunk 1]
```diff
@@ -197,6 +197,32 @@ func (c *Config) BrokerURL() *url.URL {
...
 }
 
+// BrokerDir returns the data directory to start up in and does home directory expansion if necessary.
+func (c *Config) BrokerDir() string {
+	path := c.Broker.Dir
+	u, _ := user.Current()
+	hd := u.HomeDir + "/"
+
+	// if path starts wiht ~/ we need to expand it
+	if path[:2] == "~/" {
```
Comment (otoolep) [inline cmd/influxd/config.go]: Whoa, why is this necessary? You mean if someone passes in a path like this, the OS doesn't expand it? Do you actually need this in practise?

Comment (otoolep) [inline cmd/influxd/config.go]: Can you use this instead?

http://golang.org/pkg/path/filepath/#Abs

Comment (otoolep) [inline cmd/influxd/config.go]: I'm wondering what it would do with a ~ in there.

Comment (otoolep) [inline cmd/influxd/config.go]: Look at this:
```
~ $ cat t.sh 
echo $1
~ $ bash t.sh ~/repos
/home/philip/repos
~ $ 
```

Comment (corylanou) [inline cmd/influxd/config.go]: Ah, didn't see that in the std lib when I was looking this morning.  Yes, that is a WAY WAY better way to do it.  I'll make the change.

File: cmd/influxd/run.go [hunk 2]
```diff
@@ -225,11 +225,17 @@ func openServer(path string, u *url.URL, b *messaging.Broker, initializing, conf
...
 		}
 	} else if !configExists {
-		// We are spining up an server that has no config,
+		// We are spining up a server that has no config,
 		// but already has an initialized data directory
 		joinURLs = []*url.URL{b.URL()}
 		openServerClient(s, joinURLs)
 	} else {
+		if len(joinURLs) == 0 {
+			//This is the first broker, so it always spins up a client to itself for now
```
Comment (otoolep) [inline cmd/influxd/run.go]: Nit: whitespace.

File: cmd/influxd/run.go [hunk 3]
```diff
@@ -225,11 +225,17 @@ func openServer(path string, u *url.URL, b *messaging.Broker, initializing, conf
 			openServerClient(s, joinURLs)
 		}
 	} else if !configExists {
-		// We are spining up an server that has no config,
+		// We are spining up a server that has no config,
 		// but already has an initialized data directory
 		joinURLs = []*url.URL{b.URL()}
 		openServerClient(s, joinURLs)
 	} else {
+		if len(joinURLs) == 0 {
```
Comment (otoolep) [inline cmd/influxd/run.go]: What was happening before this change?

File: cmd/influxd/config.go (hunk 1 above)
Comment (otoolep) [inline cmd/influxd/config.go]: Oh doh, this is just comes from a config file, so it's a string, not an OS thing.

Personally I think we should prohibit relative paths in config files. Allowing relative paths is a bomb waiting to go off IMHO.

File: cmd/influxd/run.go (hunk 3 above)
Comment (corylanou) [inline cmd/influxd/run.go]: It wouldn't spin up as it didn't know how to connect to a broker.  Long term this needs to come from some meta data I think, but I'm not exactly sure how we want the behavior.  For now, this fixes it but is a pretty naive approach.

File: cmd/influxd/run.go (hunk 2 above)
Comment (corylanou) [inline cmd/influxd/run.go]: I'm not sure why this does this.  Go fmt has been run, and passes.  I couldn't fix this if I wanted to.  I could move the comments above the if statement if that looks better?

Comment (otoolep) [inline cmd/influxd/run.go]: is the comment correct? The code is in `openserver()`, but the comment refers to broker.

Comment (corylanou) [inline cmd/influxd/run.go]: The comment is correct.  It is going to open a client.  Join URLs might be a naming issue that I just get confused by.  It feels like it should be "BrokerURLs", as we only join brokers right?  Even passing in "join" feels too generic to me.  This would be a bigger change though, and I could be wrong.

Comment (otoolep) [inline cmd/influxd/run.go]: Let's table the `join` naming thing for now. Perhaps I am missing something about the comment. What do you mean by "This". This what? This node? This process? The only way a process can be both a broker and a server is if it's running in single-node mode. Is that what you're saying?

Comment (corylanou) [inline cmd/influxd/run.go]: This is more due to my lack of understanding of what is possible or the design behavior yet.  I'm going to get ben to diagram some stuff one of these days!
